src/panda_sdl/frontend_sdl.cpp

```diff
@@ -92,95 +92,74 @@ void FrontendSDL::run() {
 					programRunning = false;
 					return;
 
-				case SDL_KEYDOWN:
+				case SDL_KEYDOWN: {
 					if (emu.romType == ROMType::None) break;
 
-					switch (event.key.keysym.sym) {
-						case SDLK_l: hid.pressKey(Keys::A); break;
-						case SDLK_k: hid.pressKey(Keys::B); break;
-						case SDLK_o: hid.pressKey(Keys::X); break;
-						case SDLK_i: hid.pressKey(Keys::Y); break;
-
-						case SDLK_q: hid.pressKey(Keys::L); break;
-						case SDLK_p: hid.pressKey(Keys::R); break;
-
-						case SDLK_RIGHT: hid.pressKey(Keys::Right); break;
-						case SDLK_LEFT: hid.pressKey(Keys::Left); break;
-						case SDLK_UP: hid.pressKey(Keys::Up); break;
-						case SDLK_DOWN: hid.pressKey(Keys::Down); break;
-
-						case SDLK_w:
-							hid.setCirclepadY(0x9C);
-							keyboardAnalogY = true;
-							break;
-
-						case SDLK_a:
-							hid.setCirclepadX(-0x9C);
-							keyboardAnalogX = true;
-							break;
-
-						case SDLK_s:
-							hid.setCirclepadY(-0x9C);
-							keyboardAnalogY = true;
-							break;
-
-						case SDLK_d:
-							hid.setCirclepadX(0x9C);
-							keyboardAnalogX = true;
-							break;
-
-						case SDLK_RETURN: hid.pressKey(Keys::Start); break;
-						case SDLK_BACKSPACE: hid.pressKey(Keys::Select); break;
-
-						// Use the F4 button as a hot-key to pause or resume the emulator
-						// We can't use the audio play/pause buttons because it's annoying
-						case SDLK_F4: {
-							emu.togglePause();
-							break;
+					u32 key = getMapping(event.key.keysym.scancode);
+					if (key != HID::Keys::Null) {
+						switch (key) {
+							case HID::Keys::CirclePadRight:
+								hid.setCirclepadX(0x9C);
+								keyboardAnalogX = true;
+								break;
+							case HID::Keys::CirclePadLeft:
+								hid.setCirclepadX(-0x9C);
+								keyboardAnalogX = true;
+								break;
+							case HID::Keys::CirclePadUp:
+								hid.setCirclepadY(0x9C);
+								keyboardAnalogY = true;
+								break;
+							case HID::Keys::CirclePadDown:
+								hid.setCirclepadY(-0x9C);
+								keyboardAnalogY = true;
+								break;
+							default: hid.pressKey(key); break;
 						}
-
-						// Use F5 as a reset button
-						case SDLK_F5: {
-							emu.reset(Emulator::ReloadOption::Reload);
-							break;
+					} else {
+						switch (event.key.keysym.sym) {
+							case SDLK_RETURN: hid.pressKey(Keys::Start); break;
+							case SDLK_BACKSPACE: hid.pressKey(Keys::Select); break;
```

Comment:
Ditto here

Reply:
Just man up and merge with redundant code already.

Reply:
rcow